src/graph/validator/MatchValidator.cpp

```diff
@@ -384,18 +384,28 @@ Status MatchValidator::buildColumnsForAllNamedAliases(const std::vector<QueryPar
     }
   }
 
+  std::set<std::string> visitedAliases;
   for (auto &match : currQueryPart.matchs) {
     for (auto &path : match->paths) {
```

Comment:
I think you should create a new `visitedAliases` for each query part.

Reply:
This is to avoid duplicate aliases in the `return *`. For return, only one query part, the last query part, is to be concerned. In this method, this query part is `currQueryPart`.

Within this `currQueryPart`, if there are duplicates of the same alias in multiple matches, they shall be removed, leaving only one alias in `return *`. Thus, this visitedAliases shall be declared outside this outer for loop.

Reply:
Got.